src/v2/controllers/FormController.js

```diff
@@ -59,6 +62,15 @@ export default Controller.extend({
     this.triggerAfterRenderEvent();
   },
 
+  clearMetadata() {
+    const formName = this.options.appState.get('currentFormName');
+    // if this is a terminal error, clear all transaction meta
+    if (TERMINAL_FORMS.includes(formName)) {
```

Comment:
This would mean that reloading the page would start a fresh flow instead of showing the error. Is this expected behavior? I thought that product had decided that only clicking the "cancel" (go back / signout) link would clear, and reload the page should not change anything, if there was an error, the user would still see one.

Reply:
aha, I'm not aware this requirement but I see it's good idea.
Maybe @pradeepdewda-okta / @unniravindranathan-okta could confirm.

Reply:
yeah that is correct. Keep the current state on page refresh in any case. If a terminal view doesn't have `go back` then it would be a bug to fix.

Reply:
- `FORMS.DEVICE_ENROLLMENT_TERMINAL` doesn't have sign-out link. Hopefully it's fine.
- @aarongranick-okta shall we clean transaction meta only at sign-out as well?

Reply:
the transaction meta should be cleared on "sign out", "back to sign out", or "cancel", unless cancel just means "go back". Sometimes "go back" is just returning to a previous selection menu, in this case the transaction is not being cancelled so we should not clear.

Reply:
To me for Terminal View, Go back is like starting from scratch, because remediation flow ended.

Reply:
Yes I think you are right for terminal view. "Go back" is used for some interstitial navigation but that is not on a terminal view.

Reply:
Create a follow up: OKTA-392835